lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitTask.cs

```diff
@@ -0,0 +1,227 @@
+            CleanUp(testFolder);
+        }
+
+        /// <summary>
+        /// Driver configuration for the test driver
+        /// </summary>
+        /// <returns></returns>
+        public IConfiguration DriverConfigurations()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<ResubmitTaskTestDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<ResubmitTaskTestDriver>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<ResubmitTaskTestDriver>.Class)
+                .Set(DriverConfiguration.OnTaskRunning, GenericType<ResubmitTaskTestDriver>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<ResubmitTaskTestDriver>.Class)
+                .Set(DriverConfiguration.OnTaskFailed, GenericType<ResubmitTaskTestDriver>.Class)
+                .Build();
+        }
+
+        private sealed class ResubmitTaskTestDriver :
+            IObserver<IDriverStarted>,
+            IObserver<IAllocatedEvaluator>,
+            IObserver<IActiveContext>,
+            IObserver<ICompletedTask>,
+            IObserver<IFailedTask>,
+            IObserver<IRunningTask>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+            private const string TaskId1 = "TaskId1";
+            private const string TaskId2 = "TaskId2";            
+
+            [Inject]
+            private ResubmitTaskTestDriver(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(1).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID")
+                        .Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                value.SubmitTask(GetTaskConfigurationForCloseTask(TaskId1));
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, "Task completed: " + value.Id);
+                Assert.Equal(TaskId2, value.Id);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                var failedExeption = ByteUtilities.ByteArraysToString(value.Data.Value);
+                Logger.Log(Level.Error, "In IFailedTask: " + failedExeption);
+                Assert.Equal(TaskId1, value.Id);
+                Assert.Contains(TaskKilledByDriver, failedExeption);
+                value.GetActiveContext().Value.SubmitTask(GetTaskConfigurationForCloseTask(TaskId2));
```

Comment:
You may want to cache the `ActiveContext` and do some comparison.
I'm actually a bit unsure of what the expected behavior here is -- do we have the same `ActiveContext` reference as the one that appears in `IActiveContext`? Or is this a separate `ActiveContext` instance that points to the same java `ActiveContext` instance?